### ISSUE TITLE
error from diag_puddle with gnu 7.4 compiler

### ISSUE THREAD

**Issue by trontrytel** [igfuw/libcloudphxx#374]
I'm trying to compile libcloudphxx with gnu gcc compiler 7.4.0 on Ubuntu 18.04. I get the below error. I'm not sure what to put as a return type there? 
```
/home/vagrant/libcloudphxx/include/libcloudph++/lgrngn/particles.hpp(96): warning: missing return statement at end of non-void function "libcloudphxx::lgrngn::particles_proto_t<real_t>::diag_puddle [with real_t=float]"
          detected during instantiation of "std::map<libcloudphxx::lgrngn::output_t, real_t, std::less<libcloudphxx::lgrngn::output_t>, std::allocator<std::pair<const libcloudphxx::lgrngn::output_t, real_t>>> libcloudphxx::lgrngn::particles_proto_t<real_t>::diag_puddle() [with real_t=float]" 

/home/vagrant/libcloudphxx/src/particles_diag.ipp(431): warning: missing return statement at end of non-void function "libcloudphxx::lgrngn::particles_t<real_t, backend>::diag_puddle [with real_t=float, backend=libcloudphxx::lgrngn::CUDA]"
          detected during instantiation of "std::map<libcloudphxx::lgrngn::output_t, real_t, std::less<libcloudphxx::lgrngn::output_t>, std::allocator<std::pair<const libcloudphxx::lgrngn::output_t, real_t>>> libcloudphxx::lgrngn::particles_t<real_t, backend>::diag_puddle() [with real_t=float, backend=libcloudphxx::lgrngn::CUDA]" 
/home/vagrant/libcloudphxx/src/lib_cuda.cu(18): here
```

**Comment by trontrytel** [comment]
(same as with the issue from libmpdata - I'm doing this to get the UWLCM running on the cluster in a singularity container. So I can send you the container so that the error can be easily reproduced)

**Comment by pdziekan** [comment]
PR #375 should fix this. I will accept it once it passes tests.

**Comment by pdziekan** [comment]
Merged #375 . @trontrytel please check if it fixed this issue.

**Comment by trontrytel** [comment]
yep - works. Thank you
